src/macaron/repo_finder/repo_finder_java.py

```diff
@@ -301,3 +305,46 @@ def _resolve_properties(self, pom: Element, values: list[str]) -> list[str]:
             resolved_values.append(value)
 
         return resolved_values
+
+    @staticmethod
+    def get_deps_dev_info(group: str, artifact: str, version: str) -> dict[str, Any]:
+        """Get additional project information from deps.dev for the given artifact.
+
+        info:
+          - dependentCount (how many other artifacts depend on this)
+          - repo_link (link to the repository)
+          - star_count (number of stars on the repository)
+          - fork_count (number of forks on the repository)
+        """
+        dd_info: dict[str, Any] = {}
+
+        dep_dev_url = f"https://deps.dev/_/s/maven/p/{group}%3A{artifact}/v/{version}"
+        session = requests.Session()
```

Comment:
Please use our internal [send_get_http](https://github.com/oracle/macaron/blob/edfe06eb3f9337f86215b253f40f53ca1a0105c9/src/macaron/util.py#L21) function instead. This central function ensures consistent error handling across our codebase.

Reply:
@mabdollahpour-ol It's actually better to use [send_get_http_raw](https://github.com/oracle/macaron/blob/edfe06eb3f9337f86215b253f40f53ca1a0105c9/src/macaron/util.py#L56)  function, which has a few more options and can handle rate limits.